app/src/main/java/org/covidwatch/android/ui/MainFragment.kt

```diff
@@ -1,67 +1,67 @@
 package org.covidwatch.android
 
+import android.content.Context
+import android.content.Intent
+import android.content.SharedPreferences
 import android.os.Bundle
-import androidx.fragment.app.Fragment
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import androidx.databinding.DataBindingUtil
-import androidx.fragment.app.DialogFragment
+import androidx.fragment.app.Fragment
 import androidx.navigation.findNavController
 import org.covidwatch.android.databinding.FragmentMainBinding
 
-// TODO: Rename parameter arguments, choose names that match
-// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
-private const val ARG_PARAM1 = "param1"
-private const val ARG_PARAM2 = "param2"
-
 /**
  * A simple [Fragment] subclass.
  * Use the [MainFragment.newInstance] factory method to
  * create an instance of this fragment.
  */
 class MainFragment : Fragment() {
-    // TODO: Rename and change types of parameters
-    private var param1: String? = null
-    private var param2: String? = null
-
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-        arguments?.let {
-            param1 = it.getString(ARG_PARAM1)
-            param2 = it.getString(ARG_PARAM2)
-        }
-    }
+    private var preferences : SharedPreferences? = null
+    private val INITIAL_VISIT = "INITIAL_VISIT"
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
         val binding = DataBindingUtil.inflate<FragmentMainBinding>(inflater,
             R.layout.fragment_main,container,false)
-        binding.selfReport.setOnClickListener { view : View ->
+        binding.shareTheAppButton.setOnClickListener {shareApp()}
+        binding.selfReportButton.setOnClickListener { view : View ->
             view.findNavController().navigate(R.id.action_mainFragment_to_selfReportFragment)
         }
+        preferences = activity?.getPreferences(Context.MODE_PRIVATE) ?: return binding.root
+        val initialVisit = preferences?.getBoolean(INITIAL_VISIT,true) ?: true
+        if (!initialVisit) {
+            binding.mainTitle.text = getString(R.string.welcome_back_title)
+            binding.mainText.text = getString(R.string.not_detected_text)
+        }
         return binding.root
     }
 
+    override fun onResume() {
+        super.onResume()
+        with (preferences?.edit() ?: return) {
+            putBoolean(INITIAL_VISIT, false)
+            commit()
```

Comment:
it's better to use `apply()` here to save value asynchronously

Reply:
Done.